File: lib/src/args.dart
import 'package:args/args.dart';

final argParser = ArgParser()
  ..addFlag(
    'install-drivers',
    defaultsTo: true,
    help: 'Install drivers and codecs for playing media.',
  )
  ..addFlag(
    'install-apps',
    defaultsTo: true,
    help: 'Install my favourite applications.',
  )
  ..addFlag(
    'theme-firefox',
    defaultsTo: true,
    help: 'Theme Firefox to match my COSMIC theme.',
  )
  ..addFlag(
    'theme-thunderbird',
    defaultsTo: true,
    help: 'Theme Thunderbird to match my COSMIC theme.',
  )
  ..addFlag(
    'install-firefox-cacher',
    defaultsTo: true,
    help:
        'Precache Firefox data on boot.\n'
        'Useful to speed up the first launch of Firefox.',
  )
  ..addFlag(
    'no-interaction',
    abbr: 'n',
    negatable: false,
    help: 'Skip any steps needing confirmation.',
  )
  ..addFlag(
    'yes',
    abbr: 'y',
    negatable: false,
    help: 'Always answer yes to prompts.',
  )
  ..addFlag(
    'help',
    abbr: 'h',
    negatable: false,
    help: 'Show this help message',
  );

File: src/args.rs
use clap::{ArgAction, Parser};

#[derive(Parser, Debug)]
#[command(version, about, long_about = None)]
pub struct Args {
  #[arg(long = "no-install-drivers", action = ArgAction::SetFalse, help = "Don't interactively install drivers and media codecs.")]
  pub install_drivers: bool,
  #[arg(long = "no-install-apps", action = ArgAction::SetFalse, help = "Don't interactively install my favorite applications.")]
  pub install_apps: bool,
  #[arg(long = "no-theme-firefox", action = ArgAction::SetFalse, help = "Don't theme Firefox to match your COSMIC theme.")]
  pub theme_firefox: bool,
  #[arg(long = "no-theme-thunderbird", action = ArgAction::SetFalse, help = "Don't theme Thunderbird to match your COSMIC theme.")]
  pub theme_thunderbird: bool,
  #[arg(long = "no-install-firefox-cacher", action = ArgAction::SetFalse, help = "Don't precache Firefox data on boot.\nPrecaching is useful if Firefox is one of the first things you open.")]
  pub install_firefox_cacher: bool,
}
